### PR TITLE
make additional prefileter metadata public

### PR DESCRIPTION
**Pull request by pascalkuthe**
These are the changes that I mentioned a while ago in #425

I now have a fully working streaming regex implementation at https://github.com/pascalkuthe/regex-cursor. These two APIs are the last two blockers before I can make helix switch to that implementation (we don't allow git dependencies/patches).

* makes `Prefilter::is_fast` public. Necessary to implement a meta engine downstream (also useful for any other downstream user that wants to write a custom meta engine)
* add a function that returns the maximum length of a prefilter. This is necessary to determine how much buffering is needed for streaming prefilters.

### CODE REVIEW

File: regex-automata/src/util/prefilter/mod.rs [hunk 1]
```diff
@@ -411,6 +420,19 @@ impl Prefilter {
         }
     }
 
+    /// Return the length of the longest needle
+    /// in this Prefilter
+    pub fn max_needle_len(&self) -> usize {
```
Comment (BurntSushi) [inline regex-automata/src/util/prefilter/mod.rs:425]: This should probably be marked `#[inline]`.

**Review by BurntSushi — CHANGES_REQUESTED** [review]
One little nit and I think this is good to go! Thank you!

**Comment by pascalkuthe** [comment]
done, thanks for the quick feedback!

**Review by BurntSushi — APPROVED** [review]

**Comment by BurntSushi** [comment]
This PR is on crates.io in `regex-automata 0.4.5`.

**Pull request merged**